pkg/dns/vips_allocator.go

```diff
@@ -186,6 +222,26 @@ func BuildVirtualOutboundMeshView(ctx context.Context, rm manager.ReadOnlyResour
 		}
 	}
 
+	otherMeshList := core_mesh.MeshResourceList{}
+	if err := rm.List(ctx, &otherMeshList); err != nil {
+		return nil, err
+	}
+
+	for _, otherMesh := range otherMeshList.Items {
+		otherMeshName := otherMesh.GetMeta().GetName()
+		if otherMeshName == mesh {
+			continue
+		}
+		gateways := core_mesh.MeshGatewayResourceList{}
+		if err := rm.List(ctx, &gateways, store.ListByMesh(otherMeshName)); err != nil {
+			return nil, err
+		}
+
+		for _, gateway := range gateways.Items {
+			addFromMeshGateway(outboundSet, otherMeshName, gateway)
+		}
+	}
```

Comment:
So that's the kind of loop that we'd save by controlling mesh visibility. For sure an improvement but might be useful in the case of many meshes